src/main/kotlin/org/rust/ide/inspections/RsWrongGenericArgumentsNumberInspection.kt

```diff
@@ -53,8 +61,19 @@ class RsWrongTypeArgumentsNumberInspection : RsLocalInspectionTool() {
             else -> null
         } ?: return
 
-        val problemText = "Wrong number of type arguments: expected ${errorText}, found $actualArgs"
-        val fixes = getFixes(o, actualArgs, expectedTotalParams)
+        val argumentName = when {
+            (expectedTotalTypeParams > 0 || actualTypeArgs > 0) &&
+                expectedTotalConstParams == 0 &&
+                actualConstArgs == 0 -> "type"
+            (expectedTotalConstParams > 0 || actualConstArgs > 0) &&
+                expectedTotalTypeParams == 0 &&
+                actualTypeArgs == 0 -> "const"
+            else -> "generic"
```

Comment:
Let's add `wrong number of generic arguments` tests to cover this branch as well. Currently, we have only quick fix tests with the corresponding message